tools/purge_orphans.rb

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env ruby
+require "optimist"
+
+VALID_MODES = %w[count purge].freeze
+opts = Optimist.options do
+  banner "Purge orphaned rows from several tables.\n\nUsage: ruby #{$PROGRAM_NAME} [options]\n\nOptions:\n\t"
+  opt :mode, "Mode (#{VALID_MODES.join(", ")})", :default => "count"
+end
```

Comment:
I think you need a check here that only one of the valid modes were passed. I can't remember if optimist has an enum type yet but, if so, you could use that.

Reply:
thanks, it's permitted, pushing that

```ruby
opt :mode, "Mode (#{VALID_MODES.join(", ")})", :default => "count", :permitted => VALID_MODES
```

pushed

Reply:
yes that's it - that was added recently.